src/index.js

```diff
@@ -1,110 +1,145 @@
+  const {
+    REFLECT_API_KEY,
+    REFLECT_SUITE_ID,
+    REFLECT_WAIT_FOR_TEST_RESULTS,
+    REFLECT_DEFAULT_HOSTNAME,
+    PULL_REQUEST,
+    COMMIT_REF,
+    REPOSITORY_URL,
+    DEPLOY_PRIME_URL,
+  } = process.env;
+
+  if (!REFLECT_API_KEY) {
+    return build.failPlugin("REFLECT_API_KEY is not defined.");
   }
 
-  // Console logs are shown in Netlify logs
-  console.log('Netlify configuration', netlifyConfig)
-  console.log('Plugin configuration', inputs)
-  console.log('Build directory', PUBLISH_DIR)
+  if (!REFLECT_SUITE_ID) {
+    return build.failPlugin(
+      "suite_id must be defined in Plugin configuration."
+    );
+  }
 
-  // Display success information
-  status.show({ summary: 'Success!' })
-}
+  const reflectApi = new ReflectApi(REFLECT_API_KEY);
+  let executionId;
 
-// Other available event handlers
-/*
+  try {
+    const { organization, repository } = parseRepositoryURL(REPOSITORY_URL);
+    const isPullRequest = PULL_REQUEST || PULL_REQUEST === "true";
+
+    const response = await reflectApi.executeSuite({
+      suiteId: REFLECT_SUITE_ID,
+      overrides:
+        isPullRequest && REFLECT_DEFAULT_HOSTNAME
+          ? {
+              hostnames: [
+                {
+                  original: REFLECT_DEFAULT_HOSTNAME,
+                  replacement: parseHostname(DEPLOY_PRIME_URL),
+                },
+              ],
+            }
+          : {},
+      gitHub: isPullRequest
+        ? {
+            owner: organization,
+            repo: repository,
+            sha: COMMIT_REF,
+          }
+        : {},
+    });
+    executionId = response.executionId;
+  } catch (error) {
+    return build.failPlugin(
+      `Encountered error when attempting to execute Reflect test suite: ${
+        error.message ?? "Unknown error"
+      }`,
+      { error }
+    );
+  }
 
-// Before build commands are executed
-export const onPreBuild = function () {}
+  console.log(
+    `Test suite execution located at: ${suiteExecutionUrl(
+      REFLECT_SUITE_ID,
+      executionId
+    )}`
+  );
 
-// Build commands are executed
-export const onBuild = function () {}
+  if (
+    REFLECT_WAIT_FOR_TEST_RESULTS === false ||
+    REFLECT_WAIT_FOR_TEST_RESULTS === "false"
+  ) {
+    return status.show({
+      title: `Reflect Test Suite: ${REFLECT_SUITE_ID}`,
+      summary: `Started test suite with execution id: ${executionId}.`,
+      text: `Exited without waiting for test suite to complete. See ${suiteExecutionUrl(
+        REFLECT_SUITE_ID,
+        executionId
+      )} for details.`,
+    });
+  } else {
+    console.log(
+      `Started test suite with execution id: ${executionId}. Will wait for test suite to complete before exiting.`
+    );
 
-// After Build commands are executed
-export const onPostBuild = function () {}
+    let isFinished = false;
+    let executionStatus = "";
 
-// Runs on build success
-export const onSuccess = function () {}
+    while (!isFinished) {
```

Comment:
Might want to put a cap on the retries here to avoid someone permanently spamming the API without even knowing

Reply:
My fear is that we might end up cancelling the check prematurely... It's hard to find Netlify's timeout defaults, but [this forum post](https://answers.netlify.com/t/build-timeout-after-10-minutes/19257) indicates their build timeout at the free tier is 15 minutes, but it's higher for paid accounts. I think we should let Netlify timeout the build job rather than us set a separate cap on amount of time we'll wait. Netlify also charges based on build minutes, so users have an incentive to not make their builds run a very long time.

Reply:
Works for me!